04-page-objects/src/test/java/pageobjects/Login.java

```diff
@@ -28,10 +31,19 @@ public void with(String username, String password) {
     }
 
     public Boolean successMessagePresent() {
+        wait(successMessageLocator);
         return driver.findElement(successMessageLocator).isDisplayed();
     }
 
     public Boolean failureMessagePresent() {
+        wait(failureMessageLocator);
         return driver.findElement(failureMessageLocator).isDisplayed();
     }
+
```

Comment:
Adding in explicit waits here isn't 100% necessary since we're using Chrome at this point instead of Firefox (re: document ready state bug). But, as I mentioned in chat, there is a transient timing issue that *some* (not all) attendees will experience due to how the page renders. Essentially, it's a race condition that only an explicit wait can solve (well, a hard coded sleep would solve it too, but...).

As I see it, there are two choices:

1. Incorporate the explicit wait here (like you've done) and tell people that we'll discuss this more in a later section (which is what I'm assuming you would do).

2. Leave the explicit wait function and its use in `successMessagePresent()` as you've written it, but comment it out. For people that experience the timing issue, we can instruct them to uncomment these lines, and tell them that we'll discuss this later. I like this approach (as clunky as it may sound) because it will give you an opportunity to call out obvious challenges that people will have to learn to deal with on their own. Introducing little road blocks like this is actually beneficial for instruction in my opinion.

That being said, I'm open to whatever you think. This is a good opportunity to try out some changes to the material.